gadget/gadget.go

```diff
@@ -196,12 +196,21 @@ type VolumeContent struct {
 	Size Size `yaml:"size"`
 
 	Unpack bool `yaml:"unpack"`
+
+	// XXX: we could mutate Source instead of having an extra field
+	//      but doing that leads to ugly output
+	//
+	// ResolvedSource is the absolute path of the Source after resolving
+	// any references (e.g. to a "$kernel:" snap)
+	ResolvedSource string
```

Comment:
Perhaps it would be better to instead keep Source as is and introduce a new optional string `SymbolicSource` which has the interpolation logic applied. Less changes overall perhaps?

Reply:
this PR is a bit problematic, it introduces and computes this but it doesn't make use of it yet, nor make sure that it will be used properly later.

I think we should something slightly different.  Perhaps rename Source to UnresolvedSource and have ResolvedSource as an accessor and make sure we turn Source into either, probably in most places ResolvedSource() ?